### ISSUE TITLE
option to set default lens correction method

### ISSUE THREAD

**Issue by hqhoang** [darktable-org/darktable#16724]
**Is your feature request related to a problem? Please describe.**
My Tamron 17-70/2.8 has incorrect embedded lens correction metadata, e.g. at 20mm or 31mm. I have to switch the method from "_embedded metadata_" to "_Lensfun database_" every time I apply distortion correction.

**Describe the solution you'd like**
A setting/option to set the default "correction method" for the lens correction module. The embedded data from the lens is not always correct, for instance, Tamron 17-70/2.8 for Fujifilm X.

**Comment by darkelectron** [comment]
You can create a preset with the settings of “embedded metadata”, and auto-apply that preset to your Tamron lens. You can also have it only apply to the case of: Fujifilm X + Tamron 17-70/2.8.

Source: 
- https://darktable-org.github.io/dtdocs/en/darkroom/processing-modules/presets/

**Comment by hqhoang** [comment]
Yeah, I do use auto-preset on other modules (exposure, color calibration, ...) However, I don't always apply lens correction, only about half of the time (e.g. if a shot at 17mm has people near the edges, applying distortion correction would stretch the people out weirdly).

Also, lens correction is a module where preset doesn't work. A preset is based on a specific focal length, and most shots have different focal length, thus, the correction will be incorrectly applied. Similarly, I couldn't copy/paste the lens correction from one photo to another unless the two photos are of the same focal length.

**Comment by darkelectron** [comment]
> A preset is based on a specific focal length

That is incorrect, that only applies for “auto-apply” preset.

**Comment by hqhoang** [comment]
> That is incorrect, that only applies for “auto-apply” preset.

Sigh, I'm not sure if it's a bug, but it has been that way for several years now. When I save a preset with "Lensfun database" method, the correction at that specific focal length is stored in the preset, and thus can only be applied correctly to shots at the same FL. Same problem prevents copying/pasting the lens correction from one photo to another with a different FL.

You probably can verify yourself with these steps (I'm on 4.7.0+1079~g7ebded0604-dirty):

1. edit shot  #.1 at a long focal length, e.g. 70mm in my example, apply a Lensfun distortion correction, save it as preset (e.g. "test preset at 70mm"). Note that I cannot choose what is or is not saved into the preset.
![Screenshot from 2024-05-05 14-58-31](https://github.com/darktable-org/darktable/assets/1099071/dbf04945-d1d2-4376-b41e-e79f3df116a4)

2. edit shot #.2 at a different focal length, e.g. 31mm, apply a Lensfun distortion correction, it should apply correction for 31mm.
![Screenshot from 2024-05-05 14-58-43](https://github.com/darktable-org/darktable/assets/1099071/1abc3ab8-6955-4120-a9c2-71491c856042)

3. apply the saved from preset from #.1 to #.2. Shot #.2 now has incorrect correction of 70mm applied to it.
![Screenshot from 2024-05-05 15-03-36](https://github.com/darktable-org/darktable/assets/1099071/c507e615-cf0b-4cb0-822d-f869c388e24c)

**Comment by darkelectron** [comment]
My bad, I thought you meant presets in general. 

PS: This related to #1990

**Comment by darkelectron** [comment]
@hqhoang Considering you are using the dev version, you could edit the source to make “Lensfun database” the default.

You can do this by editing the file `src/iop/lens.cc`:
1. Find the method called: `dt_iop_lens_method_t`.
2. Copy this line: `method = DT_IOP_LENS_METHOD_LENSFUN;`.
3. Comment out the if statement.
4. Paste what you copied in step 2, just before the return statement.
5. Recompile.

![20240506_021356](https://github.com/darktable-org/darktable/assets/51426724/acf6f94e-6697-4bc6-a089-a207b3397a3a)

**Comment by wpferguson** [comment]
You could solve this with a lua script.  You could invoke it with a shortcut and it could look at the values and make the correct settings.

**Comment by hqhoang** [comment]
> You could solve this with a lua script. You could invoke it with a shortcut and it could look at the values and make the correct settings.

I'm not processing a lot photos lately so it's not urgent, thus, submitting as a feature request hoping for a proper solution. A script could work, but it's just hacky temporary workaround.

**Comment by wpferguson** [comment]
> but it's just a hacky temporary workaround

The Lua scripting language is embedded in darktable.  It has access to the darktable internals and is intended for extending darktable functionality when the basic functionality isn't enough, or implementing user specific workflows.

> a proper solution

If you want to define how the solution is implemented, then you need to implement it.
